applicationset/controllers/applicationset_controller.go

```diff
@@ -1312,4 +1316,42 @@ func syncApplication(application argov1alpha1.Application, prune bool) (argov1al
 	return application, nil
 }
 
+func ownsOptions() builder.Predicates {
+	funcs := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			// if we are the owner and there is a create event, we most likely created it and do not need to
+			// re-reconcile
```

Comment:
I think there's an annotation we can use to confirm whether we created it? Or at least with more certainty.

Reply:
I didn't see any annotation on the applications created by the applicationset. Though in order for this event to be triggered someone would need to create an application with the owner reference set like this:
```
  Owner References:
    API Version:           argoproj.io/v1alpha1
    Block Owner Deletion:  true
    Controller:            true
    Kind:                  ApplicationSet
    Name:                  test-appset
    UID:                   0cac88fd-0272-4341-8398-7166d7ae1087
```

where UID is the resource ID of the applicationset controller. IMO, it is pretty unlikely but happy to be more defensive.

Reply:
Ah bet, I'm sure the ownerRef check is sufficient.